ui/cypress/e2e/sampleControls.cy.js

```diff
@@ -9,15 +9,15 @@ describe('3-click centring', () => {
 
   it('3-click centring should not work without sample', () => {
     cy.clearSamples();
-    cy.contains('button', '3-click centring').click();
+    cy.findByRole('button', { name: '3-click centring' }).click();
     cy.findByRole('alert', 'Error: There is no sample mounted').should(
       'be.visible',
     );
   });
 
   it('Each click is rotating the sample by 90 degrees', () => {
     cy.mountSample();
-    cy.contains('button', '3-click centring').click();
+    cy.findByRole('button', { name: '3-click centring' }).click();
     cy.get('.form-control[name="diffractometer.phi"]')
       .invoke('val')
       .then((value) => {
```

Comment:
While I changed most of the cypress queries to act more like a real User, there are a few input forms that do not provide a corresponding label (resp. the text that should be the label is not correctly linked). This might be investigated in future steps.